packages/core/src/test/amazonq/common/contentController.test.ts

```diff
@@ -0,0 +1,57 @@
+import * as vscode from 'vscode'
+import * as fs from 'fs-extra'
+import * as path from 'path'
+import assert from 'assert'
+import { EditorContentController } from '../../../amazonq/commons/controllers/contentController'
+import { makeTemporaryToolkitFolder } from '../../../shared'
+
+describe('contentController', () => {
+    let controller: EditorContentController
+    let tempFolder: string
+
+    beforeEach(async () => {
+        controller = new EditorContentController()
+        // Make a temp folder for all these tests
+        tempFolder = await makeTemporaryToolkitFolder()
+    })
+
+    afterEach(async function () {
+        await fs.remove(tempFolder)
+    })
+
+    describe('insertTextAtCursorPosition', () => {
+        it('insert code when left hand size has no non empty character', async () => {
+            const programFile = path.join(tempFolder, 'program.python')
+            await fs.writeFile(programFile, 'def hello_world():\n    ')
+            const doc = await vscode.workspace.openTextDocument(programFile)
+            const editor = await vscode.window.showTextDocument(doc)
```

Comment:
You can replace these first few lines with `openATextEditorWithText` from `core/src/test/testUtil`

See editorUtilities.test.ts, it has a lot of examples of using it